app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -167,6 +169,18 @@ def get_managed_images
         process_collection(images, :vms) { |image| parse_managed_image(image) }
       end
 
+      def get_market_images
+        urns   = @options.market_image_urns
+        images = if urns
+                   gather_data_for_this_region(@vmis).select do |image|
+                     urns.include?(image.id)
+                   end
+                 else
+                   gather_data_for_this_region(@vmis)
+                 end
```

Comment:
That is correct; the last time I ran this command, it gathered approximately 4500 images. I'm by no means certain if that's a reasonable default. If they just flip ```@options.market_image_urns``` to true, there are some URNs already defined to avoid that.